model/src/market/proposal.rs

```diff
@@ -48,19 +96,45 @@ impl Proposal {
     }
 
     pub fn counter_demand(&self, demand: Demand) -> Result<Proposal, ErrorMessage> {
-        Ok(Proposal {
-            properties: demand.properties,
-            constraints: demand.constraints,
-            proposal_id: None,
-            issuer_id: None,
-            state: None,
-            prev_proposal_id: Some(self.proposal_id()?.clone()),
-        })
+        let mut properties = demand.properties.clone();
+        let demand_platforms = parse_payment_platforms(
+            &properties
+                .get("golem")
+                .unwrap_or(&serde_json::json!({}))
+                .get("com")
+                .unwrap_or(&serde_json::json!({})),
```

Comment:
In many places I see code that parses Agreement and it seems to be a lot of pain.
Do you know, that `AgreementView` struct exists, that has nice utils, that could help??

Reply:
you might use https://crates.io/crates/ya-agreement-utils